sleyster_a07b/part_1/js/script.js

```diff
@@ -0,0 +1,143 @@
+            loadingImageContainer.classList.add('centered');
+            container.appendChild(loadingImageContainer);
+            }
+        }
+    
+    function clearLoading(){
+        right.innerHTML = '';
+    }
+
+    function checkAirportCode(){
+        const airportCodeValue = airportCode.value;
+        if (airportCodeValue == '') return false;
+        if (airportCodeValue.length > 3) return false;
+        // Begin Step 3
+        //created array to store valid codes then itterated through to validate entry.
+        if(airportCodeValue.length < 3) return false;
+        const validCodes = ['anw', 'bvn', 'aia', 'auh', 'bie', 'bta', 'hde', 'bbw', 'cdr', 
+            'olu', 'fnb', 'fet', 'gri', 'hsi', 'hjh', 'iml', 'ear', 'ibm', 'lxn', 'lnk',
+            'mck', 'mle', 'afk', 'lbf', 'onl', 'oga', 'off', 'oma', 'odx', 'pmv', 'bff',
+            'sny', 'tqe', 'tif', 'vtn', 'ahq', 'lcg', 'jyr'];
+        for(const validCode of validCodes) {
+            airportCodeValue.toLowerCase();
+            if(validCodes.includes(airportCodeValue)) return true;
+            return false;
+        }
+        //End Step 3
+        return true; 
+    }
+
+    function displayData(xml){
+        let parser = new DOMParser();
+        let xmlDoc = parser.parseFromString(xml, 'text/xml');
+        console.log(xmlDoc);
+        let location = xmlDoc.getElementsByTagName('location')[0].innerHTML;
+        let temp_f = xmlDoc.getElementsByTagName('temp_f')[0].innerHTML;
+        //Begin Step 4
+        /*got elements by their tage name from the xml document and created elements
+        in the html document and appended the information to the html elements.
+        */
+        let temp_c = xmlDoc.getElementsByTagName('temp_c')[0].innerHTML;
+        //let windchill_f = xmlDoc.getElementsByTagName('windchill_f')[0].innerHTML;
+        //let windchill_c = xmlDoc.getElementsByTagName('windchill_c')[0].innerHTML;
+        let visibility_mi = xmlDoc.getElementsByTagName('visibility_mi')[0].innerHTML;
+        let wind_mph = xmlDoc.getElementsByTagName('wind_mph')[0].innerHTML;
+        let h1 = document.createElement('h1');
+        let cw = `Current Weather`;
+        h1.innerHTML = cw;
+        right.appendChild(h1);
+        let h2 = document.createElement('h2');
+        h2.innerHTML = location;
+        right.appendChild(h2);
+        let p = document.createElement('p');
+        let pText = `Temperature: ${temp_f} ${2109}(${temp_c} ${2103}) \n
+                    Wind Speed: ${wind_mph} MPH \n
+                    Visiblilty: ${visibility_mi}`;
+        p.innerHTML = pText;
+        right.appendChild(p);
+        //commented out the original output for new output to show.
+        //let wx = `The current temperature in ${location} is ${temp_f} F`;
+        //right.innerHTML = wx;
+        //End Step 4 
+        //Begin Step 5
+        const icon_url = 'http://forcast.weather.gov/images/wtf/small';
```

Comment:
you are going to need to take this address, add the code for the current conditions (icon name - I think) and build the URL like you did with the airport code. You can that use that URL to build an image to add to the output. - that is step 5 & 6.